cmake/engine_git.cmake

```diff
@@ -1,50 +1,83 @@
-function (set_git_version BRANCH HASH VERSION)
+function (set_git_version BRANCH HASH VERSION_UNFORMATTED)
     set(OBENGINE_GIT_BRANCH ${BRANCH} CACHE STRING "ObEngine git branch" FORCE)
     set(OBENGINE_GIT_COMMIT_HASH ${HASH} CACHE STRING "ObEngine git commit hash" FORCE)
-    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version" FORCE)
+
+    string(REGEX MATCH "v(.+)\\.(.+)\\.(.+)$" VERSION "${VERSION_UNFORMATTED}")
+
+    set(OBENGINE_VERSION ${VERSION} CACHE STRING "ObEngine version string" FORCE)
+    set(VERSION_MAJOR ${CMAKE_MATCH_1})
+    set(VERSION_MINOR ${CMAKE_MATCH_2})
+    set(VERSION_PATCH ${CMAKE_MATCH_3})
+
+    string(REGEX MATCH "([0-9]+)$" RES ${VERSION_MAJOR})
+    if(NOT ${RES} STREQUAL "")
+        set(VERSION_MAJOR ${CMAKE_MATCH_1})
+        # 0*([0-9]+)$ seems nicer, but CMake can not lazily match
+        math(EXPR VERSION_MAJOR "${VERSION_MAJOR} * 1")
+    else()
+        set(VERSION_MAJOR "-1")
+    endif()
+
+    string(REGEX MATCH "([0-9]+)$" RES ${VERSION_MINOR})
+    if(NOT ${RES} STREQUAL "")
+        set(VERSION_MINOR ${CMAKE_MATCH_1})
+        math(EXPR VERSION_MINOR "${VERSION_MINOR} * 1")
+    else()
+        set(VERSION_MINOR "-1")
+    endif()
+
+    string(REGEX MATCH "([0-9]+)$" RES ${VERSION_PATCH})
+    if(NOT ${RES} STREQUAL "")
+        set(VERSION_PATCH ${CMAKE_MATCH_1})
+        math(EXPR VERSION_PATCH "${VERSION_PATCH} * 1")
+    else()
+        set(VERSION_PATCH "-1")
+    endif()
+
+    set(OBENGINE_VERSION_MAJOR ${VERSION_MAJOR} CACHE STRING "ObEngine major version" FORCE)
+    set(OBENGINE_VERSION_MINOR ${VERSION_MINOR} CACHE STRING "ObEngine minor version" FORCE)
+    set(OBENGINE_VERSION_PATCH ${VERSION_PATCH} CACHE STRING "ObEngine patch version" FORCE)
 endfunction()
 
-function (configure_obengine_git)
-    find_program(GIT_FOUND git)
-    message("Configure Git variables for ÖbEngine")
-    if (GIT_FOUND)
+function (configure_obengine_git GIT_EXECUTABLE ObEngine_SOURCE_DIR)
+    message(STATUS "Configure Git variables for ÖbEngine")
+    if (NOT GIT_EXECUTABLE STREQUAL "")
         execute_process(
-            COMMAND git rev-parse --is-inside-work-tree
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            COMMAND ${GIT_EXECUTABLE} rev-parse --is-inside-work-tree
+            WORKING_DIRECTORY ${ObEngine_SOURCE_DIR}
```

Comment:
Isn't ${ObEngine_SOURCE_DIR} a global variable ? Do we have to pass it as parameter ?

Reply:
This script is not included in the main CMakeLists.txt anymore but rather executed at build-time. Therefore some parameters have to be manually propagated.
When using "configure_file" the files will only be updated during configure time. If you create a new commit and build your project, the changes will not be reflected in the build unless you reconfigure the CMake project.

Reply:
Ah ! Makes sense :)